### ISSUE TITLE
Allow decodedJwt.getPayload().getTree() as HashMap

### ISSUE THREAD

**Issue by djangofan** [auth0/java-jwt#368]
When trying to follow the instructions in your documentation on decoding a JWT, I am noticing the lack of exposed methods to get the decoded payload as a Map.

Example:  I would like to do:
```
     DecodedJWT decodedJwt = JWT.decode(jwt);
     HashMap<String, Object> map = decodedJwt.getPayload().getTree()
```
NOTE: where getPayload() perhaps would return a object, not a String

It's not clear to me why your APi is not making `getTree()` method public/available.  Maybe there is a reason for it or another way to get it?

**Comment by lbalmaceda** [comment]
@djangofan There are [plenty of issues already closed](https://github.com/auth0/java-jwt/issues?q=is%3Aissue+json+object+map+is%3Aclosed) for this feature request.  

**TL;DR:** Not currently supported, we look into adding something with interfaces (extendable by the end-user) but we don't have the bandwidth to do it today.

However, if you happen to know the claim name, you can obtain that one as Map in the following way:

```
Map<String, Object> userDataMap = jwt.getClaim("user-data").asMap();
```
